lib/libzfs/libzfs_pool.c

```diff
@@ -3588,6 +3588,16 @@ zpool_vdev_name(libzfs_handle_t *hdl, zpool_handle_t *zhp, nvlist_t *nv,
 	char buf[PATH_BUF_LEN];
 	char tmpbuf[PATH_BUF_LEN];
 
+	/*
+	 * vdev_name will be "root"/"root-0" for the root vdev, but it is the
+	 * zpool name that will be displayed to the user.
+	 *
+	 * TODO: figure out a way to get the zpool name when zhp is NULL
+	 */
+	verify(nvlist_lookup_string(nv, ZPOOL_CONFIG_TYPE, &type) == 0);
+	if (strcmp(type, "root") == 0 && zhp != NULL)
```

Comment:
nit: can you invert this so the NULL check comes before the strcmp().

Reply:
Good idea, fixed.